File: src/daily_report/utils/io.py
from __future__ import annotations

import json
from pathlib import Path
import yaml


def ensure_dir(path: Path) -> Path:
    path.mkdir(parents=True, exist_ok=True)
    return path


def write_json(path: Path, data) -> None:
    ensure_dir(path.parent)
    path.write_text(json.dumps(data, ensure_ascii=False, indent=2), encoding="utf-8")


def write_yaml(path: Path, data) -> None:
    ensure_dir(path.parent)
    path.write_text(yaml.safe_dump(data, allow_unicode=True, sort_keys=False), encoding="utf-8")


def read_yaml(path: Path):
    return yaml.safe_load(path.read_text(encoding="utf-8"))


def write_text(path: Path, text: str) -> None:
    ensure_dir(path.parent)
    path.write_text(text, encoding="utf-8")

File: src/daily_report_js/utils/io.js
import fs from 'node:fs';
import path from 'node:path';
import yaml from 'js-yaml';

export function ensureDir(dirPath) {
  fs.mkdirSync(dirPath, { recursive: true });
}

export function readYaml(filePath) {
  return yaml.load(fs.readFileSync(filePath, 'utf-8')) || {};
}

export function writeJson(filePath, data) {
  ensureDir(path.dirname(filePath));
  fs.writeFileSync(filePath, `${JSON.stringify(data, null, 2)}\n`, 'utf-8');
}

export function writeYaml(filePath, data) {
  ensureDir(path.dirname(filePath));
  fs.writeFileSync(filePath, yaml.dump(data, { sortKeys: false, noRefs: true }), 'utf-8');
}

export function writeText(filePath, text) {
  ensureDir(path.dirname(filePath));
  fs.writeFileSync(filePath, text, 'utf-8');
}

export function loadJson(filePath) {
  return JSON.parse(fs.readFileSync(filePath, 'utf-8'));
}

export function exists(filePath) {
  return fs.existsSync(filePath);
}

export function listDir(dirPath) {
  return fs.existsSync(dirPath) ? fs.readdirSync(dirPath) : [];
}
